PlayChat/AppDelegate.swift

```diff
@@ -14,208 +14,216 @@
+    textField.isUserInteractionEnabled = true
+    textField.textColor = UIColor.white
+    textField.backgroundColor = UIColor.black
     textField.becomeFirstResponder()
     textField.delegate = self
     channelView.view.addSubview(textField)
-    
+
     return channelView
   }
-  
-  func application(application: UIApplication, openURL url: NSURL,
-                   options: [String: AnyObject]) -> Bool {
-    return GIDSignIn.sharedInstance().handleURL(url, sourceApplication:
-      options[UIApplicationOpenURLOptionsSourceApplicationKey] as? String,
-              annotation: options[UIApplicationOpenURLOptionsAnnotationKey])
+
+  func application(
+    _ application: UIApplication,
+    open url: URL,
+    options: [UIApplicationOpenURLOptionsKey: Any]
+  ) -> Bool {
+    return GIDSignIn.sharedInstance().handle(
+      url,
+      sourceApplication: options[UIApplicationOpenURLOptionsKey.sourceApplication] as? String,
+      annotation: [: ]
+    )
   }
-  func applicationWillResignActive(application: UIApplication) {}
-  func applicationDidEnterBackground(application: UIApplication) {}
-  func applicationWillEnterForeground(application: UIApplication) {}
-  func applicationDidBecomeActive(application: UIApplication) {}
-  func applicationWillTerminate(application: UIApplication) {}
-  
-  func signIn(signIn: GIDSignIn!, didSignInForUser user: GIDGoogleUser!,
-              withError error: NSError!) {
+  func applicationWillResignActive(_ application: UIApplication) {}
+  func applicationDidEnterBackground(_ application: UIApplication) {}
+  func applicationWillEnterForeground(_ application: UIApplication) {}
+  func applicationDidBecomeActive(_ application: UIApplication) {}
+  func applicationWillTerminate(_ application: UIApplication) {}
+
+  func sign(_ signIn: GIDSignIn, didSignInFor user: GIDGoogleUser, withError error: Error?) {
     if let error = error {
       print("signIn error : \(error.localizedDescription)")
       return
     }
-    
-    if (self.user == nil) {
+
+    if self.user == nil {
       self.user = user
-      let authentication = user.authentication
-      let credential =
-        FIRGoogleAuthProvider.credentialWithIDToken(
-          authentication.idToken,
-          accessToken: authentication.accessToken)
-      FIRAuth.auth()?.signInWithCredential(credential) { (user, error) in
-        print("Signed-in to Firebase as \(user!.displayName!)")
+      guard let authentication = user.authentication else { return }
+      let credential = GoogleAuthProvider.credential(withIDToken: authentication.idToken,
+                                                     accessToken: authentication.accessToken)
+      Auth.auth().signInAndRetrieveData(with: credential) { (authResult, error) in
+          if error != nil {
+              return
+          }
+        print("Signed-in to Firebase as \(String(describing: authResult?.user.displayName))")
```

Comment:
`authResult` should be nonnull here, so you can force-unwrap it.

Reply:
Done.